fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/common/StoreTimer.java

```diff
@@ -389,7 +389,13 @@ public interface Count extends StoreTimer.Event {
 
     /**
      * {@link Event}s that count the number of occurrences of an operation and also the size of the operation.
-     * There is no meaningful time duration associated with these events.
+     * {@link SizeEvent} is similar in implementation to (Time){@link Event} except for the fact that the recorded value
+     * is not the time. It is also different from {@link Count} since along with the size of the event in concern, it
+     * also records the number of times the record has occurred.
+     *
+     * This is in principle similar to having 2 {@link Count} event like, {@code COUNT_NUM} and {@code COUNT_SIZE} with
+     * {@link Count#isSize()} is equal to true for the latter. However, in this case the consumer ends up registering 2
+     * events in case of an occurrence and also maintains how they are interpreted by viewers.
```

Comment:
I feel like I _kind of_ understand what this is trying to say, but it's a bit muddled. I think you're trying to say that by combining the count and size into a single event, it becomes easier for the user to manage because they only need to record a single occurrence to update both and means that the link between the total and count is more obvious when reading metrics. It may need some word smithing if that's what you mean